### PR TITLE
Add "inf" and "NaN" as valid nodata values

### PR DESCRIPTION
**Pull request by emmanuelmathot**
Allow "inf" and "nan" as valid nodata values #18

### CODE REVIEW

**Review by vincentsarago — APPROVED** [review]

**Comment by emmanuelmathot** [comment]
from semantic versioning point of view, is this a breaking change?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Allow "inf" and "NaN" as valid nodata values [#18](https://github.com/stac-extensions/raster/issues/18)
```
Comment (lossyrob) [inline CHANGELOG.md:11]: Just to be consistent with the json schema value...
```suggestion
- Allow "inf" and "nan" as valid nodata values [#18](https://github.com/stac-extensions/raster/issues/18)
```

**Comment by lossyrob** [comment]
That's a good question...if the schema change made things that _were_ valid to be _not_ valid, I would say definitely yes. But anything that was valid before is still valid, it just broadens the scope of what is a valid nodata value...

I think having a bugfix version update would be appropriate here as it's not breaking for valid items and it is a sort of bug fix.

**Review by lossyrob — APPROVED** [review]
Approve after minor inline change for consistency

**Comment by emmanuelmathot** [comment]
> I think having a bugfix version update would be appropriate here as it's not breaking for valid items and it is a sort of bug fix.

I would say MINOR version because we add functionality in a backwards compatible manner

File: json-schema/schema.json [hunk 2]
```diff
@@ -132,7 +132,19 @@
...
+            "oneOf": [
+              {
+                "type": "number"
+              },
+              {
+                "type": "string",
+                "enum": [
+                  "nan",
+                  "inf",
+                  "-inf"
```
Comment (vincentsarago) [inline json-schema/schema.json:144]: @emmanuelmathot I may have approved this to soon 

```python
# in python
son.dumps({"v": numpy.nan})
>>> '{"v": NaN}'

json.dumps({"v": -numpy.inf})
>>> '{"v": -Infinity}'

json.dumps({"v": numpy.inf})
>>> '{"v": Infinity}'
``` 

worth noting that in node, Infinity and NaN are number, so maybe we could have let the schema to number 🤷‍♂️ 
```node
typeof(Infinity)
>>> 'number'

typeof(NaN)
>>> 'number'
```

cc @lossyrob

Comment (emmanuelmathot) [reply json-schema/schema.json:144]: the problem is not python that seems to be handling the JSON special number issue on its own. The main issue is the interoperability. If you submit a JSON with `{"v": Infinity}` to C# default JSON reader, it raises an error.

Comment (emmanuelmathot) [reply json-schema/schema.json:144]: e.g. in vs code
image

Comment (vincentsarago) [reply json-schema/schema.json:144]: 👍 I'll make sure that my code encode `NaN` to `"nan"` and `Inf` to `inf` then 👍

**Pull request merged**